File: VB/WindowsApplication1/my_grid_control.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Columns

Namespace WindowsApplication1
	Public Class MyGridControl
		Inherits GridControl
		Public Sub New()
			MyBase.New()
		End Sub
		Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
			MyBase.RegisterAvailableViewsCore(collection)
			collection.Add(New MyGridInfoRegistrator())
		End Sub
	End Class

	Public Class MyGridView
		Inherits GridView
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal control As GridControl)
			MyBase.New(control)
		End Sub
		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property
		Protected Overrides Function CreateCheckedFilterPopup(ByVal column As GridColumn, ByVal ownerControl As Control, ByVal creator As Object) As CheckedColumnFilterPopup
			Return New MyCheckedColumnFilterPopup(Me, column, ownerControl, creator)
		End Function
	End Class

	Public Class MyGridInfoRegistrator
		Inherits GridInfoRegistrator
		Public Sub New()
			MyBase.New()
		End Sub
		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyGridView(grid)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication1/my_grid_control.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;

namespace WindowsApplication1 {
    public class MyGridControl : GridControl {
        public MyGridControl() : base() { }
        protected override void RegisterAvailableViewsCore(InfoCollection collection) {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyGridInfoRegistrator());
        }
    }

    public class MyGridView : GridView {
        public MyGridView() : base() { }
        public MyGridView(GridControl control) : base(control) { }
        protected override string ViewName { get { return "MyGridView"; } }
        protected override CheckedColumnFilterPopup CreateCheckedFilterPopup(GridColumn column, Control ownerControl, object creator) { 
            return new MyCheckedColumnFilterPopup(this, column, ownerControl, creator); 
        }
    }

    public class MyGridInfoRegistrator : GridInfoRegistrator {
        public MyGridInfoRegistrator() : base() { }
        public override string ViewName { get { return "MyGridView"; } }
        public override BaseView CreateView(GridControl grid) { return new MyGridView(grid); }
    }
}